static/signup.js

```diff
@@ -0,0 +1,34 @@
+function signupDefault() {
+    let username = document.getElementById("username").value;
+    let email = document.getElementById("email").value;
+    let givenName = document.getElementById("given_name").value;
+    let familyName = document.getElementById("family_name").value;
+    let password = document.getElementById("password").value;
+
+    // Encrypt the data, then send it to the server
+    let payload = {
```

Comment:
I would recommend to avoid storing sensitive data into variables, extract them directly and save into the dictionary variables for encryption, i.e.: username: document.getElementById("username").value;